src/raft/raft.rs

```diff
@@ -304,6 +304,11 @@ impl<T: Storage> Raft<T> {
         hs
     }
 
+    pub fn in_lease(&self, safe_distance: usize) -> bool {
+        self.state == StateRole::Leader && self.check_quorum &&
+        self.election_elapsed + safe_distance <= self.election_timeout
```

Comment:
now we can think time is ok. we will consider time problem after @swingbach and us think a solution.